src/session/session_dhandle.c

```diff
@@ -438,7 +438,7 @@ __wt_session_lock_checkpoint(WT_SESSION_IMPL *session, const char *checkpoint)
 	 * files, since changes to the underlying file are visible to the in
 	 * memory pages.
 	 */
-	WT_ERR(__wt_cache_op(session, NULL, WT_SYNC_DISCARD));
+	WT_ERR(__wt_cache_op(session, NULL, WT_SYNC_CLOSE));
```

Comment:
I'll revert.  What I discovered is that `WT_SYNC_CLOSE` isn't safe to call outside of `__wt_checkpoint` because it asserts that `ckptbase` has been set up when it tries to evict the root.